### ISSUE TITLE
The tensorflow evaluate result is never correct

### ISSUE THREAD

**Issue by LeonMac** [adap/flower#1398]
### Describe the bug

By using this small sample: https://github.com/adap/flower/tree/main/examples/quickstart_tensorflow.
Need some upgrades on both client/server to reflect the flwr1.0.0 change.
then running the test, it shows the evaluation result is always kept around 10%.
```
get dataset cifar10: train sample size: 50000, test sample size: 10000
train sample size: 50000, test sample size: 10000
<...>
DEBUG flower 2022-09-01 10:46:08,783 | server.py:220 | fit_round 1: strategy sampled 2 clients (out of 2)
20/20 [==============================] - 5s 96ms/step - loss: 2.8006 - accuracy: 0.1164
20/20 [==============================] - 5s 97ms/step - loss: 2.8174 - accuracy: 0.1211
DEBUG flower 2022-09-01 10:46:13,738 | server.py:234 | fit_round 1 received 2 results and 0 failures
WARNING flower 2022-09-01 10:46:13,834 | fedavg.py:243 | No fit_metrics_aggregation_fn provided
DEBUG flower 2022-09-01 10:46:13,834 | server.py:170 | evaluate_round 1: strategy sampled 2 clients (out of 2)
313/313 [==============================] - 4s 10ms/step - loss: 2.3026 - accuracy: 0.1000
313/313 [==============================] - 4s 10ms/step - loss: 2.3026 - accuracy: 0.1000
DEBUG flower 2022-09-01 10:46:17,589 | server.py:184 | evaluate_round 1 received 2 results and 0 failures
WARNING flower 2022-09-01 10:46:17,589 | fedavg.py:274 | No evaluate_metrics_aggregation_fn provided
DEBUG flower 2022-09-01 10:46:17,589 | server.py:220 | fit_round 2: strategy sampled 2 clients (out of 2)
20/20 [==============================] - 2s 99ms/step - loss: 2.5599 - accuracy: 0.1320
20/20 [==============================] - 2s 99ms/step - loss: 2.5569 - accuracy: 0.1352
DEBUG flower 2022-09-01 10:46:19,974 | server.py:234 | fit_round 2 received 2 results and 0 failures
DEBUG flower 2022-09-01 10:46:20,059 | server.py:170 | evaluate_round 2: strategy sampled 2 clients (out of 2)
313/313 [==============================] - 3s 10ms/step - loss: 2.3026 - accuracy: 0.1000
313/313 [==============================] - 3s 10ms/step - loss: 2.3026 - accuracy: 0.1000

<...>
```

### Steps/Code to Reproduce

server code:
```
import flwr as fl

RND = 5
# Start Flower server
fl.server.start_server(
    server_address="[::]:8080",
    config = fl.server.ServerConfig( num_rounds=RND ),
    )

```
client code:
```
import os

import flwr as fl
import tensorflow as tf


# Make TensorFlow log less verbose
os.environ["TF_CPP_MIN_LOG_LEVEL"] = "3"

# Load model and data (MobileNetV2, CIFAR-10)
model = tf.keras.applications.MobileNetV2((32, 32, 3), classes=10, weights=None)
model.compile("adam", "sparse_categorical_crossentropy", metrics=["accuracy"])
(x_train, y_train), (x_test, y_test) = tf.keras.datasets.cifar10.load_data()

# Define Flower client
class CifarClient(fl.client.NumPyClient):
    def get_parameters(self, config):
        return model.get_weights()

    def fit(self, parameters, config):
        model.set_weights(parameters)
        model.fit(x_train, y_train, epochs=1, batch_size=64, steps_per_epoch=20)
        return model.get_weights(), len(x_train), {}

    def evaluate(self, parameters, config):
        model.set_weights(parameters)
        loss, accuracy = model.evaluate(x_test, y_test)
        return loss, len(x_test), {"accuracy": accuracy}


# Start Flower client
fl.client.start_numpy_client(server_address = "localhost:8080", client=CifarClient())
```

### Expected Results

The evaluation result shown should be better and better as we run more training rounds

### Actual Results

The evaluation result (loss/accuracy) keeps at the same number, regardless of how many rounds of the training were run.

**Comment by LeonMac** [comment]
Hello, I wonder if anyone had tried to reproduce the issue by using the above code? I have been working on PyTorch and Sklearn as well but did not notice such an issue, which is weird -- this is not likely the server issue but a client issue, however, I didn't find any code error in the client. 
My only suspicion is the parameters that are updated to evaluate(self, parameters, config), is not correct (or never been updated)

**Comment by WilliamLindskog** [comment]
Hi, 

Thanks for raising this issue and clarifying some points. I am closing this as it seems a TF issue and the flwr code in this example has been updated.
